python_modules/dagster/dagster/_core/definitions/job_definition.py

```diff
@@ -106,193 +141,379 @@ def __init__(
         _subset_selection_data: Optional[Union[OpSelectionData, AssetSelectionData]] = None,
         asset_layer: Optional[AssetLayer] = None,
         input_values: Optional[Mapping[str, object]] = None,
-        _executor_def_specified: Optional[bool] = None,
-        _logger_defs_specified: Optional[bool] = None,
-        _preset_defs: Optional[Sequence[PresetDefinition]] = None,
         _was_explicitly_provided_resources: Optional[bool] = None,
     ):
         from dagster._core.definitions.run_config import RunConfig, convert_config_input
-        from dagster._loggers import default_loggers
 
-        check.inst_param(graph_def, "graph_def", GraphDefinition)
-        resource_defs = check.opt_mapping_param(
-            resource_defs, "resource_defs", key_type=str, value_type=ResourceDefinition
-        )
-        # We need to check whether an actual executor/logger def was passed in
-        # before we set a default executor/logger defs. This is so we can
-        # determine if someone passed in the default executor vs the system set
-        # it directly. Once JobDefinition no longer subclasses
-        # PipelineDefinition, we can change the default executor to be set
-        # elsewhere to avoid the need for this check.
-        self._executor_def_specified = (
-            _executor_def_specified
-            if _executor_def_specified is not None
-            else executor_def is not None
-        )
-        self._logger_defs_specified = (
-            _logger_defs_specified
-            if _logger_defs_specified is not None
-            else logger_defs is not None
-        )
-        executor_def = check.opt_inst_param(
-            executor_def, "executor_def", ExecutorDefinition, default=multi_or_in_process_executor
-        )
-        check.opt_mapping_param(
```

Comment:
Todo: review

Reply:
subtle behavior change in that subsequent calls to `with_executor_def` don't preserve whether executor / loggers were originally specified on the job, but realistically I think that's fine.